lib/set.rb

```diff
@@ -521,9 +521,12 @@ def add(o)
   #     Set[1, 2].add?([3, 4])               #=> #<Set: {1, 2, [3, 4]}>
   #     Set[1, 2].add?(2)                    #=> nil
   def add?(o)
-    add(o) unless include?(o)
+    self unless @hash.update_value(o, true)
```

Comment:
This alternative implementation had promising benchmark results. We should investigate it, too.

```ruby
def add?(o)
  (@hash.update_value(o, true) || nil) && self
end
```

It's really unreadable, though :D